sw/device/lib/dif/dif_rstmgr.c

```diff
@@ -10,84 +10,70 @@
 #include "sw/device/lib/base/mmio.h"
 #include "rstmgr_regs.h"  // Generated.
 
-_Static_assert(kDifRstmgrCausePowerOn == RSTMGR_RESET_INFO_CAUSE_POR,
+// This macro simplifies the `_Static_assert` check to make sure that the
+// public reset info register bitfield matches register bits.
+#define RSTMGR_CAUSE_CHECK(pub_name, priv_name) \
+  kDifRstmgrResetInfo##pub_name == (0x1 << RSTMGR_RESET_##priv_name)
```

Comment:
Can you also make this generate the error message? E.g.
```
_Static_assert(..., "kDifRstmgrRestInfo" #pub_name " must match the register definition!");
```

Reply:
Good point.

Done.